cmd/snap/cmd_ack.go

```diff
@@ -52,6 +52,10 @@ func init() {
 }
 
 func (x *cmdAck) Execute(args []string) error {
+	if len(args) > 0 {
```

Comment:
so one different approach to this would be in main.go, to change

builder to just  

`
   builder func() interface{}
`
then have  a CommanderNoArgs interface

```
    type CommanderNoArgs {
       ExecuteNoArgs() error
    }

```
and then when we take
    
`    obj := c.builder()
`

we check, if it implements flags.Commander we just pass it along,
if it implements CommanderNoArgs we wrap it into:

```
    type commanderNoArgs struct {
       CommanderNoArgs
    }

    func (c commanderNoArgs) Execute(args []string) error {
      if len(args) > 0 {
       ...
      } 
      return c.ExecuteNoArgs()
    }

```